src/Model.cc

```diff
@@ -792,11 +796,34 @@ const NestedInclude *Model::InterfaceModelNestedIncludeByIndex(
 }
 
 /////////////////////////////////////////////////
-sdf::ElementPtr Model::ToElement() const
+std::string Model::Uri() const
+{
+  return this->dataPtr->uri;
+}
+
+/////////////////////////////////////////////////
+void Model::SetUri(const std::string &_uri)
+{
+  this->dataPtr->uri = _uri;
+}
+
+/////////////////////////////////////////////////
+sdf::ElementPtr Model::ToElement(bool _useUri) const
 {
+  if (_useUri && !this->dataPtr->uri.empty())
+  {
+    sdf::ElementPtr worldElem(new sdf::Element);
+    sdf::initFile("world.sdf", worldElem);
+
+    sdf::ElementPtr includeElem = worldElem->AddElement("include");
+    includeElem->GetElement("uri")->Set(this->Uri());
+    includeElem->GetElement("name")->Set(this->Name());
+    includeElem->GetElement("pose")->Set(this->RawPose());
```

Comment:
I think there are other attributes of pose that we need to copy: `relative_to`, `degrees`, `rotation_format`.

Reply:
`relative_to` checks in b3ca02a. I think the `degrees` and `rotation_format` are not parsed by the DOM at all and should be tackled in  another PR.

Reply:
Ah, right, never mind about `degrees` and `rotation_format`.